lib/tictactoe/cli/console.ex

```diff
@@ -0,0 +1,50 @@
+defmodule TicTacToe.CLI.Console do
+  def display_instructions do
+    IO.puts("Welcome to Tic Tac Toe!\n")
+    IO.puts(instructions())
+    IO.puts("Starting new game...\n")
+  end
+
+  def display_board(board) do
+    length = row_length(board.cells)
+
+    board.cells
+    |> Enum.chunk_every(length)
+    |> Enum.map_join(horizontal_line(length), &format_row(&1))
+    |> IO.puts()
+  end
+
+  defp row_length(cells) do
+    cells
+    |> length()
+    |> :math.sqrt()
+    |> trunc()
+  end
+
+  defp format_row(row_array) do
+    row_array
+    |> Enum.map_join("|", &format_cell(&1))
```

Comment:
It's typically preferred to use the pipe operator when working with multiple calls (there's even a [credo rule](https://hexdocs.pm/credo/Credo.Check.Readability.SinglePipe.html) for it). In this case, it would be a bit easier to read:

```elixir
Enum.map_join(row_array, "|", &format_cell(&1))
```

Reply:
Ah, sorry for the confusion. What I meant was that in this case, the pipeline is only one function long, therefore we should eliminate the pipe operator and write out the function call normally. I think we can still combine the `map` and `join` functions, but just whenever you see yourself doing a single pipe chain like this, then prefer to just write a function call.

Reply:
Ahhh ok I gotcha... thanks for clarifying!